src/main/java/de/dhbw/karlsruhe/students/mailflow/core/external/email/EmailMetadataFactory.java

```diff
@@ -0,0 +1,72 @@
+package de.dhbw.karlsruhe.students.mailflow.core.external.email;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.Address;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.EmailMetadata;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.Header;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.Recipients;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.SentDate;
+import de.dhbw.karlsruhe.students.mailflow.core.domain.email.value_objects.Subject;
+import jakarta.mail.Message;
+import jakarta.mail.MessagingException;
+import jakarta.mail.Message.RecipientType;
+
+public class EmailMetadataFactory {
+    private Message message;
+
+    public EmailMetadataFactory(Message message) {
+        this.message = message;
+    }
+
+    public static EmailMetadataFactory withMessage(Message message) {
+        EmailMetadataFactory factory = new EmailMetadataFactory(message);
+        return factory;
+    }
+
+    private List<Address> getRecipients(RecipientType recipientType) throws MessagingException {
+        List<Address> addresses = new ArrayList<>();
+
+        jakarta.mail.Address[] recipients = message.getRecipients(recipientType);
+
+        if (recipients == null || recipients.length == 0) {
+            return Collections.emptyList();
+        }
```

Comment:
the `recipients == null` check is necesary to avoid a NPE, the `recipients.length == 0` is just a early return

Reply:
What I mean is, you could implement this functionality in the Email aggregate and this functionality inside the domain. I think its part of the domain logic to return an emptyList when no recipients exist.

Anyways, this implementation would need a bit more to be adjusted, because here we are working with jakarta emails but we could do this with our domain emails. We can change it afterwards if you want. Also I am not 100% sure if it works the way I imagine it